### PR TITLE
S2U-42 Roster: Card Game - Make card game group aware

### PR DESCRIPTION
**Pull request by stetsche**
Only display students to learn in the that are in the same  group

https://sakaiproject.atlassian.net/browse/S2U-42

### CODE REVIEW

**Review by mpellicer — CHANGES_REQUESTED** [review]
I don't think this works for sites without groups, to be honest, can you try a plain site without groups with the patch?

**Comment by jesusmmp** [comment]
These changes group aware are OK. Tested on my local instance.
Let me test a site without groups

**Comment by mpellicer** [comment]
> These changes group aware are OK. Tested on my local instance. Let me test a site without groups

I think it fails in site cases. Is this for TAs? Can TAs see all the site members? Not talking about content, I mean MEMBERS.

If an instructor is a member of the site, does not matter how many groups are there, it should see all the students, at the end 
this feature must display the same users that site-manage does.

**Comment by mpellicer** [comment]
What do you feel about this @jesusmmp ?

**Review by mpellicer — APPROVED** [review]
This sounds nice to me, I think this replicates existing logic in the roster.

**Comment by jesusmmp** [comment]
> What do you feel about this @jesusmmp ?

Let me test it! :)

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/CardGameController.java [hunk 1]
```diff
@@ -135,6 +141,52 @@ public ResponseEntity<String> reset() {
         return ResponseEntity.ok().build();
     }
 
+    private Set<String> getVisibleUsersIds(String userId, Site site) {
+        if (StringUtils.isBlank(userId) || site == null) {
+            return Collections.emptySet();
+        }
+
+        String siteId = site.getId();
```
Comment (jesusmmp) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/CardGameController.java:149]: This siteId should be /site/ + site.getId();

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Let's test all of this in our QA, in sites without groups and sites with groups and TAs.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/CardGameController.java [hunk 2]
```diff
@@ -135,6 +143,67 @@ public ResponseEntity<String> reset() {
...
+        Set<String> userIds;
+        if (securityService.unlock(userId, ROSTER_PERM_VIEW_ALL_MEMBERS, siteRef)) {
+            userIds = site.getMembers().stream()
+                    .filter(memberFilter)
+                    .map(Member::getUserId)
+                    .collect(Collectors.toSet());
+
+            log.info("view all; users: {}", userIds.toArray());
+            if (!securityService.unlock(userId, ROSTER_PERM_VIEW_HIDDEN, siteRef)) {
+                Set<String> hiddenUsersIds = privacyManager.findHidden(site.getReference(), userIds);
```
Comment (jesusmmp) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/CardGameController.java:178]: Use siteRef

**Pull request merged**